### PR TITLE
use redux in react

### PR DESCRIPTION
**Pull request by vanina123**
Includes an array of books (initial state: empty array)
Includes a reducer that adds a book
Includes a reducer that removes a book
Define a slice of state for categories that:
Includes an array of categories (initial state: empty array)
Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)
Check linter errors
Use correct gitflows

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,153 @@
...
+- [⭐️ Show your support](#support)
+- [🙏 Acknowledgements](#acknowledgements)
+- [❓ FAQ (OPTIONAL)](#faq)
+- [📝 License](#license)
+
+
+# 📖 Book Store <a name="about-project"></a>
+
+
+**Math Magician** is a webapp built with React that contain books with authors and their categories
```
Comment (FrederickMih) [inline README.md]: - [Optional] Please kindly update this description. This is a bookstore app and not Math Magician app

File: package.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
+{
+  "name": "books-store",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@reduxjs/toolkit": "^1.9.5",
```
Comment (FrederickMih) [inline package.json:6]: - [ ] Great job for adding redux toolkit for state management. I think you equally need to add react-redux to effectively set up redux store in your app. I couldn't see this dependency. You can install it using `npm install react-redux` or `yarn add react-redux`.

File: src/redux/store.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { createStore, combineReducers } from '@reduxjs/toolkit';
```
Comment (FrederickMih) [inline src/redux/store.js]: - [ ] As stated in the requirements, let's kindly utilize configureStore to create a reducer. We will require to, first of all, import it from @reduxjs/toolkit

File: src/redux/store.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+import booksSlice from './books/booksSlice';
+import categoriesSlice from './categories/categoriesSlice';
+
+const rootReducer = combineReducers({
```
Comment (FrederickMih) [inline src/redux/store.js]: - [ ] As stated above, let's kindly use configureStore to create reducers.

File: src/redux/categories/categoriesSlice.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = 'Under Construction';
```
Comment (FrederickMih) [inline src/redux/categories/categoriesSlice.js]: - [ ] Let's kindly adhere to the project requirements. Initialstate should have a state value `categories` that contain an empty array and not a string. It should be something like:

```.js
initialState: {
  categories: [ ] #empty array
}
```

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand:  @vanina123, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- PR is professional :+1: 


## Required Changes ♻️

- Below :arrow_down: 

_Check the comments under the review._

## Optional suggestions

- Below :arrow_down: 

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand:  @vanina123, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- PR is professional :+1: 


## Required Changes ♻️

- Below :arrow_down: 

_Check the comments under the review._

## Optional suggestions

- Below :arrow_down: 

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: package.json (hunk 2 above)
Comment (tugsem) [reply package.json:6]: @vanina123  As the reviewer requested, please add react-redux to your project dependencies.

**Review by tugsem — APPROVED** [review]
## STATUS: APPROVED 👍🎯  
Hello  @vanina123 👋 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⭐ 
✔️ configureStore is used
✔️ react-redux is installed
✔️ description is updated
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**